istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,361 @@
+			}
+
+			if err = createConfig(kubeClient, wg, clusterID, revision, outputName); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	generateConfigCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
+	generateConfigCmd.PersistentFlags().StringVarP(&outputName, "output", "o", "", "Name of the tarball to be created")
+
+	generateConfigCmd.PersistentFlags().StringVar(&clusterID, "cluster id", "", "The ID used to identify the cluster")
+	generateConfigCmd.PersistentFlags().StringVar(&revision, "revision", "", "control plane revision (experimental)")
+	return generateConfigCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputName string) error {
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, temp); err != nil {
+		return err
+	}
+	if err = createCertificates(kubeClient, temp); err != nil {
+		return err
+	}
+	if err = createMeshConfig(kubeClient, wg, clusterID, revision, temp); err != nil {
+		return err
+	}
+	if err = Tar(temp, outputName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	we := wg.Spec.Template.Spec
+	for _, v := range we.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": strings.Join(ports, ","),
+		"ISTIO_NAMESPACE":     wg.Namespace,
+		"ISTIO_PILOT_PORT":    "15012",
+		"ISTIO_SERVICE":       fmt.Sprintf("%s.%s", wg.Name, wg.Namespace),
+		"ISTIO_SERVICE_CIDR":  "*",
+		"SERVICE_ACCOUNT":     we.ServiceAccount,
+	}
+	return ioutil.WriteFile(dir+"/cluster.env", []byte(mapToString(clusterEnv)), tempPerms)
+}
+
+// Get and store the needed certificates
+// TODO: user internal generate-cert
+func createCertificates(kubeClient kube.ExtendedClient, dir string) error {
+	rootCert, err := kubeClient.CoreV1().ConfigMaps("istio-system").Get(context.Background(), "istio-ca-root-cert", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if err = ioutil.WriteFile(dir+"/root-cert.pem", []byte(rootCert.Data["root-cert.pem"]), tempPerms); err != nil {
+		return err
+	}
+
+	args := strings.Split("run istio.io/istio/security/tools/generate_cert -client -host spiffee://cluster.local/vm/vmname --out-priv key.pem --out-cert cert-chain.pem -mode citadel", " ")
+	cmd := exec.Command("go", args...)
```

Comment:
No. We cannot depend on golang to be present on the host system in order to run a command, even if it is experimental. Please sort out how to integrate generate_cert - possibly as a pkg into istioctl.